invertedai/common.py

```diff
@@ -141,22 +141,35 @@ class AgentAttributes(BaseModel):
     AgentState
     """
 
-    length: float  #: Longitudinal extent of the agent, in meters.
-    width: float  #: Lateral extent of the agent, in meters.
+    length: Optional[float] = None  #: Longitudinal extent of the agent, in meters.
+    width: Optional[float] = None  #: Lateral extent of the agent, in meters.
     #: Distance from the agent's center to its rear axis in meters. Determines motion constraints.
-    rear_axis_offset: float
+    rear_axis_offset: Optional[float] = None
+    agent_type: Optional[str] = 'car'  #: Type of agent, default to car, can also be pedestrian
 
     @classmethod
     def fromlist(cls, l):
-        length, width, rear_axis_offset = l
-        return cls(length=length, width=width, rear_axis_offset=rear_axis_offset)
+        if len(l) == 4:
+            length, width, rear_axis_offset, agent_type = l
+            return cls(length=length, width=width, rear_axis_offset=rear_axis_offset, agent_type=agent_type)
+        else:
+            agent_type = l[0]
```

Comment:
If you're asserting that `l` only has one element, `agent_type, = l` would be preferable. If it can have more, I'm a little worried about how you know that the agent type comes first.